tasks/build.py

```diff
@@ -346,20 +348,94 @@ def download_pr(repo_name, branch_name, pr, arch_job_dir):
+        )
+    if clone_exit_code != 0:
+        error_stage = 'git clone'
+        return clone_output, clone_error, clone_exit_code, error_stage
 
     git_checkout_cmd = ' '.join([
         'git checkout',
         branch_name,
     ])
-    checkout_output, checkout_err, checkout_exit_code = run_cmd(git_checkout_cmd,
-                                                                "checkout branch '%s'" % branch_name, arch_job_dir)
+    log(f'checking out with command {git_checkout_cmd}')
+    checkout_output, checkout_err, checkout_exit_code = run_cmd(
+        git_checkout_cmd, "checkout branch '%s'" % branch_name, arch_job_dir, raise_on_error=False
+        )
+    if checkout_exit_code != 0:
+        error_stage = 'git checkout'
+        return checkout_output, checkout_err, checkout_exit_code, error_stage
 
     curl_cmd = f'curl -L https://github.com/{repo_name}/pull/{pr.number}.diff > {pr.number}.diff'
-    curl_output, curl_error, curl_exit_code = run_cmd(curl_cmd, "Obtain patch", arch_job_dir)
+    log(f'curl with command {curl_cmd}')
+    curl_output, curl_error, curl_exit_code = run_cmd(
+        curl_cmd, "Obtain patch", arch_job_dir, raise_on_error=False
+        )
+    if curl_exit_code != 0:
+        error_stage = "curl"
+        return curl_output, curl_error, curl_exit_code, error_stage
 
     git_apply_cmd = f'git apply {pr.number}.diff'
-    git_apply_output, git_apply_error, git_apply_exit_code = run_cmd(git_apply_cmd, "Apply patch", arch_job_dir)
+    log(f'git apply with command {git_apply_cmd}')
+    git_apply_output, git_apply_error, git_apply_exit_code = run_cmd(
+        git_apply_cmd, "Apply patch", arch_job_dir, raise_on_error=False
+        )
+    if git_apply_exit_code != 0:
+        error_stage = 'git apply'
+        return git_apply_output, git_apply_error, git_apply_exit_code, error_stage
+
+
+def comment_download_pr(base_repo_name, pr, download_pr_exit_code, download_pr_error, error_stage):
+    """
+    Handle download_pr() exit code and write helpful comment to PR in case of failure
+
+    Args:
+        base_repo_name (string): name of the repository (format USER_OR_ORGANISATION/REPOSITORY)
+        pr (github.PullRequest.PullRequest): instance representing the pull request
+        download_pr_exit_code (int): exit code from download_pr(). 0 if all tasks were successful,
+            otherwise it corresponds to the error codes of git clone, git checkout, git apply, or curl.
+        download_pr_error (string): none, or the output of stderr from git clone, git checkout, git apply or curl.
+        error_stage (string): a string informing the stage where download_pr() failed. Can be 'git clone',
+            'git checkout', 'curl', or 'git apply'.
+
+    Return:
+        None (implicitly). A comment is created in the appropriate PR.
+
+    """
+    if download_pr_exit_code != 0:
+        fn = sys._getframe().f_code.co_name
+        if error_stage == 'git clone':
```

Comment:
Use globally defined variable `ERROR_GIT_CLONE`, e.g.,
```suggestion
        if error_stage == ERROR_GIT_CLONE:
```

Reply:
Moved string constants to globally defined variables in header.